Missing visualization management actions

dmfalke opened the pull request:
Add ability to name and delete visualizations. Closes #23 .

bobular commented on package.json:
```diff
@@ -14,6 +14,7 @@
     "react-dom": "^17.0.1",
     "uuid": "^8.3.2"
   },
+  "sideEffects": false,
```
Curious what this is for...

moontrip replied:
@bobular  I was curious about that so I googled it to learn about the option :) Although you may find it easily, here is a summary. It is related to Webpack's optimization, so-called tree shaking. Simply put, Webpack tries to exclude unused external library/package to minimize its compiled file size (optimization), but certainly it often makes codes not work correctly, so-called side effects. The optimization is performed when the sideEffects is set to false (meaning side effects are free), so any side effects by Webpack's optimization become developers' responsibility. Actually, the option can be set at a library's package.json too.

Certainly it is more complex than what I said earlier: please see https://webpack.js.org/guides/tree-shaking/ if you are interested

bobular reviewed (commented):
I haven't been through all the code - it's my day off!  But I do have some general observations/suggestions after playing with the checked-out code for a while

- After clicking "Copy session" the new session name could be "<Old session name> copy"
- The visualizations could also have a copy button (with same behaviour as above)
- While in subsetting tab, "Copy session" creates new session, back at the default subsetting variable (see also #35 )
- While in viz tab, "Copy session" creates a "Visualization not found." message on the screen.  However, if I switch to subsetting and back to viz, the visualizations were copied just fine.
- The grid view viz's should show the viz name.

bobular commented:
PS. the delete session button in the actual session page (not the session list) seems non-operational.

dmfalke commented:
Thanks for the feedback, @bobular. I think I am going to incorporate addressing those issues in this PR. I will make github issues for those. I might also include the variable picker updates in this PR. I haven't decided yet.

dmfalke commented:
I've merged the latest from `main` into this branch. I think it's ready for review.

dmfalke commented:
> I haven't been through all the code - it's my day off! But I do have some general observations/suggestions after playing with the checked-out code for a while
> 
>     * After clicking "Copy session" the new session name could be `<Old session name> copy`
> 
>     * The visualizations could also have a copy button (with same behaviour as above)
> 
>     * While in subsetting tab, "Copy session" creates new session, back at the default subsetting variable (see also #35 )
> 
>     * While in viz tab, "Copy session" creates a "Visualization not found." message on the screen.  However, if I switch to subsetting and back to viz, the visualizations were copied just fine.
> 
>     * The grid view viz's should show the viz name.

@bobular I just realized I forgot about these issues you raise. I will address these soon. I guess I will put this back in draft mode for now.

dmfalke commented:
@bobular I've addressed the issues you raised. This is ready for review now. Thanks!

bobular commented:
Looks great. Functionally I can't find any issues - and noted the analysis actions are also working nicely :-)

I'm not sure that the viz's should be shown in most-recently-edited first order.  Seems to be any config edit (title or plot settings).  This clearly isn't urgent for the upcoming demo.  Shall I make a ticket for this?   I guess a discussion on what grid view really is (is it a proto-dashboard?) would be useful first.  Should the grid items be re-orderable by dragging for example?

In terms of aesthetics and UX, I'll just make a couple of comments, even though I'm sure it's going to get a proper look at in a separate ticket

1. expanding a plot in grid view should simply involve clicking on it (I know this depends on some plot-components props standardisation by me, e.g. for switching off mouse-over tooltips that are still active on Histogram)
2. in grid-view, the copy and delete icons could appear on mouse-over of the minimised viz item (no icons shown when not-moused over)
3. in fullscreen view, you probably already know this, the copy and minimise icons are visually similar/confusing.

Also for another ticket/mockup meeting, font sizes and overall page layout could be made clearer. (For example, the study name is prefixed with "Study:" but the analysis and viz aren't.)  No idea what the solution is though (not my strong suit).
![image](https://user-images.githubusercontent.com/308639/119493950-a94b7b00-bd58-11eb-9c80-0587a4ed6a8d.png)

dmfalke commented:
I am going to merge this. I will make issues for things discussed here.
